### PR TITLE
br/stream: allow pitr to create oversized indices

### LINKED ISSUE #58430: PiTR: cannot restore index created during log backup if it is longer than `3072`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Start a cluster with:

```fish
tiup playground nightly --tiflash=0 --db.config=./op/adhoc/index-length-8001.toml
```

Where the config `index-length-8001.toml` is:

```toml
max-index-length = 8001
```

Do the following:

```fish
mycli --port 4000 --execute "DROP TABLE IF EXISTS test.huge_idx;"
mycli --port 4000 --execute "CREATE TABLE test.huge_idx(id int AUTO_INCREMENT, blob1 varchar(1000), blob2 varchar(1000))"

$br backup full -s (s3path "$strg_prefix-full") $argv
set backupts (mcli cat "loc/breeze/$strg_prefix-full/backupmeta" | _get_brpb BackupMeta | rg -r '$1' 'end_version: (\d+)$')
$br log start -s (s3path "$strg_prefix-incr") --task-name test $argv --start-ts "$backupts"

mycli --port 4000 --execute "CREATE INDEX huge ON test.huge_idx(blob1, blob2);"

# Wait until the checkpoint advances...
$br restore point -s (s3path huge_index-incr) --full-backup-storage (s3path huge_index-full)
```

### 2. What did you expect to see? (Required)

Restore should success.

### 3. What did you see instead (Required)

```
repair ingest index huge for table test.huge_idx  ABORTED
["restore log failed summary"] [error="failed to repair ingest index: [ddl:1071]Specified key was too long (4000 bytes); max key length is 3072 bytes"]
```

### 4. What is your TiDB version? (Required)

nightly

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #58430 

Problem Summary:
During restoring, the internal config of BR wasn't updated. 
It may reject all indices longer than the default config (`3072`) though the target cluster modified this config.

### What changed and how does it work?
This updates the TiDB config of the BR internal TiDB instance during restoring incremental data.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fixed a bug that caused cannot restore index created during log backup if it is longer than `3072`.
```

### CODE REVIEW

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -1321,6 +1321,9 @@ func restoreStream(
 	ctx, cancelFn := context.WithCancel(c)
 	defer cancelFn()
 
+	restoreCfg := enableTiDBConfig()
```
Comment (Tristan1900) [inline br/pkg/task/stream.go:1324]: maybe let's put a comment here for what it's used for?

**Review by Tristan1900 — APPROVED** [review]

File: br/pkg/task/stream.go (hunk 1 above)
Comment (YuJuncen) [reply br/pkg/task/stream.go:1324]: I renamed the function. Perhaps its name now presents its intention properly?

**Review by BornChanger — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**